### ISSUE TITLE
Dynamics per track

### ISSUE THREAD

**Issue by dgriessler** [CoderLine/alphaTab#341]
# Description 

You can now specify dynamics in AlphaTex and they last until you change them which is awesome. I would like to specify different dynamics per track. Per staff is extra bonus.

# Possible Solutions

Isolate the remembered dynamic to only its track.

# Current Workarounds 

None.

**Comment by Danielku15** [comment]
I am not sure if I get you right on your request, can you provide a concerete example with what you mean? 

With the latest changes alphaTex remembers the last used dynamics value for all next beats and you can change it any time by specifying it on the beat. Also when starting new tracks and staffs. 

The only thing I could imagine you mean is that the dynamics should reset to `F` when you start a new staff/track. This would eliminate the need that a user needs to reset it to `F` manually.
Is that what you mean or should there be some special feature where you can somehow specify dynamics differently than today?

**Comment by dgriessler** [comment]
As a specific example, consider when adding tracks for multiple instruments or when adding 4 tracks for a choir and 1 track for a piano. They aren't necessarily playing/singing at the same dynamic level. So for four of the tracks, I need mp for the first 8 bars but the piano needs to show f for the first 8 bars. Currently, when I specify mp in at least one track then all of the tracks say mp but that isn't the desired behavior.

**Comment by Danielku15** [comment]
What I meant is: Can you phrase out a concrete example _for alphaTab_. I could not understand yet what you are asking for. Please sketch out a proposal with alphaTex to highlight in a concrete case what the situation is and what you'd like to have. 

In your mentioned example you would simply specify the dynamics accordingly at the start of the track: 

```
\track "voice 1"
1.1.4{dy mp} | 1.1 | 1.1 {dy f} | 1.1

\track "voice 2"
1.1{dy mp} | 1.1 | 1.1 | 1.1
                
\track "paino"
1.1{dy f} | 1.1 | 1.1 | 1.1    
```

![image](https://user-images.githubusercontent.com/674916/74608385-d1bde180-50e0-11ea-900f-be408d408e5a.png)

I see the point that it might not make always sense to "remember" the dynamics across tracks. Resetting it to default `F` might make more sense. But I am not yet sure if this is really the point you are highlighting. You mentioned there is no workaround to your issue and this particular case is easily solveable. So I guess you mean something else.

**Comment by dgriessler** [comment]
I see what you're asking for now thank you for phrasing it again.  All that I noticed was that when I specified mp for one track then suddenly all of the tracks were mp so I didn't try and set some of the other tracks to different dynamics as you did in your example.  Overall, I would expect when I specify a track with a dynamic level mp that it only persists in that track and does not impact the other tracks i.e. dynamic control and memory of last dynamic level is confined to each individual track. 

I was wrong when I said there was no workaround as your example clearly demonstrates so my apologies.

For example, consider the following example. I would expect that the correct behavior would be the first measure of the first track has a dynamic level of mp, the second measure of the first track has a dynamic level of mf and the 2nd track has the default dynamic level of f. However, the current behavior is to process track by track and remember the last dynamic level processed so the 1st measure of the 2nd track was set to mf since the last dynamic level processed was mf in the 1st track.

`\track "Soprano 1"
\\ts 6 8 r.2{d dy mp} | r.2{d dy mf} |

\track "Soprano 2"
\\ts 6 8 r.2{d} | r.2{d} |`

![image](https://user-images.githubusercontent.com/32967199/74611181-9a8f0680-50c7-11ea-987c-861517337106.png)

My suggestion, therefore, is to reset the dynamic level to the default at the beginning of a new track which I realize now is kinda pedantic since as you've shown you just need to specify a beginning dynamic level for each track and you get the behavior you want.
